### PR TITLE
Improve doc for HttpClient

### PR DESCRIPTION
**Pull request by nicolas-grekas**
While in the plane from SymfonyLive Sao Paulo :)

### CODE REVIEW

File: reference/configuration/framework.rst [hunk 1]
```diff
@@ -662,39 +683,49 @@ when the request starts with this path.
...
+                    auth_bearer: secret_bearer_token
                     # ...
 
+Options defined for scoped clients apply only to URLs that match either their
+`base_uri`_ or the `scope`_ option when it is defined. Non-matching URLs always
+use default options.
+
+Each scoped clients also define corresponding named autowiring aliases.
+E.g. if you use
+``Symfony\\Constracts\\HttpClient\\HttpClientInterface $myApiClient``
```
Comment (OskarStark) [inline reference/configuration/framework.rst]: In double backticks, one backslash is enough

File: components/http_client.rst [hunk 2]
```diff
@@ -335,19 +342,15 @@ class to autoconfigure the HTTP client based on the requested URL::
     use Symfony\Component\HttpClient\ScopingHttpClient;
 
     $client = HttpClient::create();
-    $httpClient = new ScopingHttpClient($client, [
-        // the key is a regexp which must match the beginning of the request URL
+    $client = new ScopingHttpClient($client, [
+        // the options definesd as values apply only to the URLs matching
```
Comment (javiereguiluz) [inline components/http_client.rst]: `definesd` -> `defined`

File: components/http_client.rst [hunk 3]
```diff
@@ -362,30 +365,26 @@ regular expression applied to relative URLs::
...
-However, it includes the :class:`Symfony\\Component\\HttpClient\\Psr18Client`
+This component uses and implements abstractions defined by the
+``symfony/contracts`` package. It also implements the `PSR-18`_ (HTTP Client)
+specifications via the :class:`Symfony\\Component\\HttpClient\\Psr18Client`
 class, which is an adapter to turn a Symfony ``HttpClientInterface`` into a
 PSR-18 ``ClientInterface``.
 
-Before using it in your application, run the following commands to install the
-required dependencies:
+To use it, you the ``psr/http-client`` package and use a `PSR-17`_
```
Comment (javiereguiluz) [inline components/http_client.rst]: This reads weird -> `To use it, you the [...]`  I think the original phrase was better in this case.

File: reference/configuration/framework.rst [hunk 4]
```diff
@@ -662,39 +683,49 @@ when the request starts with this path.
...
+            scoped_clients:
+                my_api.client:
+                    auth_bearer: secret_bearer_token
                     # ...
 
+Options defined for scoped clients apply only to URLs that match either their
+`base_uri`_ or the `scope`_ option when it is defined. Non-matching URLs always
+use default options.
+
+Each scoped clients also define corresponding named autowiring aliases.
```
Comment (javiereguiluz) [inline reference/configuration/framework.rst]: `clients` -> `client`

**Review by javiereguiluz — COMMENTED** [review]
Very nice improvements! Thank you Nicolas!!

**Comment by nicolas-grekas** [comment]
First comments addressed, ready for merge if you're OK with fixing the TODOs later.

**Comment by noniagriconomie** [comment]
Just asking, is there some benchmarks comparing this httpclient with others php ones? Thank you

**Comment by OskarStark** [comment]
@noniagriconomie I found this recently: https://twitter.com/jderusse/status/1115333801419399170

**Comment by noniagriconomie** [comment]
@OskarStark yes i found it to once, but i mean real benchmarks

**Comment by javiereguiluz** [comment]
Thank you Nicolas.

File: reference/configuration/framework.rst [hunk 5]
```diff
@@ -662,39 +683,49 @@ when the request starts with this path.
 http_client
 ~~~~~~~~~~~
 
-If there's only one HTTP client defined in the app, you can configure it
-directly under the ``framework.http_client`` option:
+When the HttpClient component is installed, an HTTP client is available
+as a service named ``http_client`` or using the autowiring alias
+:class:`Symfony\\Constracts\\HttpClient\\HttpClientInterface`.
```
Comment (nicolas-grekas) [inline reference/configuration/framework.rst]: `s/Constracts/Contracts`

**Pull request merged**